fsm/fsm.go

```diff
@@ -0,0 +1,46 @@
+package fsm
+
+type State int
```

Comment:
Let's migrate out `State struct` over to here so we can have nicely named states.